src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/BOM/MaterialForm/MaterialFormView.tsx

```diff
@@ -202,23 +204,26 @@ const MaterialFormView: React.FC<MaterialFormViewProps> = ({
                     variant="outlined"
                     error={!!errors.unitName}
                     helperText={errors.unitName?.message}
+                    value={field.value || ''}
+                    onChange={(event) => {
+                      const selectedValue = event.target.value;
+                      if (selectedValue === 'createUnit') {
+                        const unitName = prompt('Enter Unit Name');
+                        if (unitName) {
+                          createUnit(unitName);
+                        } else {
+                          event.target.value = field.value === undefined ? '' : field.value;
```

Comment:
So if you click on + Create Unit and then hit cancel, and then try to submit the form it crashes. I believe its because this line changes the unit name from "undefined" to "" and then the same bug that you had with pdmFileName happens. Should be easy to fix (knock on wood)